src/source/amplify.rs

```diff
@@ -16,13 +60,13 @@ where
 #[derive(Clone, Debug)]
 pub struct Amplify<I> {
     input: I,
-    factor: f32,
+    factor: AmpFactor,
 }
 
 impl<I> Amplify<I> {
     /// Modifies the amplification factor.
     #[inline]
-    pub fn set_factor(&mut self, factor: f32) {
+    pub fn set_factor(&mut self, factor: AmpFactor) {
```

Comment:
alternative design: do not use `AmpFactor` but keep this a `f32`, then add a second method to `Amplify` for setting a decibel factor. For example: `set_log_factor`.